install/kubernetes/cilium/README.md

```diff
@@ -352,5 +352,6 @@ contributors across the globe, there is almost always someone available to help.
 | tls | object | `{"enabled":true,"secretsBackend":"local"}` | Configure TLS configuration in the agent. |
 | tolerations | list | `[{"operator":"Exists"}]` | Node tolerations for agent scheduling to nodes with taints ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/ |
 | tunnel | string | `"vxlan"` | Configure the encapsulation configuration for communication between nodes. Possible values:   - disabled   - vxlan (default)   - geneve |
+| tunnelPort | int | `8472` (VXLAN), `6081` (Geneve) | Configure the tunnel port to use |
```

Comment:
@bmcustodio helm-docs doesn't seem to like this, what do you think I should do here? I want to document this value, but I don't want to set it to zero by default... neither I want to have separate values for VXLAN and Geneve, that just doesn't seem to make sense.

Reply:
I am not entirely sure what you mean, but looking at https://github.com/cilium/cilium/pull/16874/files#diff-83e9b8b2ccacddb2fbdb28fbcfa148473d05056c0b4c955fc971709b469d7331R355 I would say you want to add `tunnelPort` as a value to `values.yaml` and document it there:

```yaml
# -- Configure the tunnel port to use.
# By default, this is the default port for VXLAN (the default value for `.tunnel`).
# You may want to set this value to 6081 if using Geneve instead.
tunnelPort: 8472
```

Reply:
My intension was to leave the defaulting to the agent, I am avoid duplicating it in the Helm template.

Reply:
That makes sense, but then I would not add `tunnelPort` to the chart — if it's up to the agent to figure out, then it doesn't _need_ to be in the configmap right? If that's the case, I would just leave `hasKey` then but keep it as a "hidden" value. We do that already for some other values AFAIU.

Reply:
Ok, it's probably fine to keep this hidden, right now the use-case for it is to do with OpenShift, and we can document it in OpenShift docs.